### PR TITLE
#404/create post and delete user roles eps

### PR DESCRIPTION
**Pull request by Damrod**
as described in #404

### CODE REVIEW

File: transcendence-app/src/auth/auth.controller.ts [hunk 1]
```diff
@@ -93,4 +103,32 @@ export class AuthController {
   loginLocalUser(@Req() req: Request, @Body() user: LoginUserDto): User {
     return req.user;
   }
+
+  @Post('local/role')
```
Comment (gbudau) [inline transcendence-app/src/auth/auth.controller.ts:107]: praise: I like how you implemented these endpoints

question: What do you think about moving the role routes to the authorization controller?

`local/role` -> `authorization/role`

File: transcendence-app/src/authorization/infrastructure/db/user-to-role.entity.ts [hunk 2]
```diff
@@ -11,11 +12,19 @@ export interface UserToRoleData {
...
-  id: string;
-  role: Role;
+  @IsString()
+  @IsUUID()
+  @IsNotEmpty()
+  id!: string;
+
+  @IsEnum(Role)
+  @IsNotEmpty()
+  role!: Role;
```
Comment (gbudau) [inline transcendence-app/src/authorization/infrastructure/db/user-to-role.entity.ts:22]: suggestion: We could remove the `@IsNotEmpty` and `@IsString` decorators

question: What do you think about creating a `UserToRoleDto`?

**Review by gbudau — APPROVED** [review]
Thank you for this feature! LGTM :rocket:

File: transcendence-app/src/auth/auth.controller.ts (hunk 1 above)
Comment (Damrod) [reply transcendence-app/src/auth/auth.controller.ts:107]: Yes, why not.

File: transcendence-app/src/authorization/infrastructure/db/user-to-role.entity.ts (hunk 2 above)
Comment (Damrod) [reply transcendence-app/src/authorization/infrastructure/db/user-to-role.entity.ts]: I don't like duplicated code, and UserToRole Dto would be the third object with that shape, and the second object that implements the usertoroledata interface. But I guess it's more readable if it exists, and so I have made the change you requested.

**Pull request merged**